src/interval.jl

```diff
@@ -279,22 +279,24 @@ end
 
 # There is power in a union.
 """
-    union(intervals::AbstractVector{<:AbstractInterval})
+    union(intervals::Union{AbstractInterval, AbstractVector{<:AbstractInterval}}, itrs...)
 
-Flattens a vector of overlapping intervals into a new, smaller vector containing only
+Flattens sets of overlapping intervals into a new, smaller vector containing only
 non-overlapping intervals.
 """
-function Base.union(intervals::AbstractVector{<:AbstractInterval})
-    return union!(convert(Vector{AbstractInterval}, intervals))
+function Base.union(
+    interval::Union{AbstractInterval, AbstractVector{<:AbstractInterval}}, intervals...,
+)
+    return union!(AbstractInterval[interval; intervals...])
 end
 
-"""
-    union!(intervals::AbstractVector{<:Union{Interval, AbstractInterval}})
+function Base.union!(intervals::AbstractVector{T}) where T <: AbstractInterval
+    if !(T <: Interval) && T != AbstractInterval
+        throw(ArgumentError(
+            "Cannot `union!` array of intervals of type $T as the type may change"
+        ))
+    end
```

Comment:
It would be better to use dispatch here for future interval types can support their own `union!` methods
```julia
function Base.union!(intervals::AbstractVector{<:AbstractInterval}) = 
    throw(ArgumentError("Cannot `union!` array of intervals of type $T as the type may change"))
end

Base.union!(intervals::AbstractVector{<:Union{AbstractInterval,Interval}) = ...
```
(I think this works)

Reply:
Doesn't the existing definition already allow that?

Reply:
I think my point here was that this approach in the PR isn't the best for new interval types that want to define their own `union!` method.